src/test/java/com/amazonaws/encryptionsdk/DefaultCryptoMaterialsManagerTest.java

```diff
@@ -308,4 +423,31 @@ public void decrypt_whenTrailingSigMissing_throwsException() throws Exception {
             }
         }
     }
+
+    @Test
+    public void decrypt_whenTrailingSigMissingForKeyring_throwsException() {
+        for (CryptoAlgorithm algorithm : CryptoAlgorithm.values()) {
+            if (algorithm.getTrailingSignatureLength() == 0) {
+                continue;
+            }
+
+            EncryptionMaterials encryptMaterials = easyGenMaterialsForKeyring(
+                    builder -> builder.setRequestedAlgorithm(algorithm)
+            );
+
+            DecryptionMaterialsRequest request = DecryptionMaterialsRequest.newBuilder()
+                    .setEncryptedDataKeys(encryptMaterials.getEncryptedDataKeys())
+                    .setAlgorithm(algorithm)
+                    .setEncryptionContext(Collections.emptyMap())
+                    .build();
+
+            try {
+                new DefaultCryptoMaterialsManager(keyring1).decryptMaterials(request);
+                fail("expected exception");
```

Comment:
assertThrows for new code at least please :)

Reply:
I'd been avoiding using JUnit5 assertions in JUnit 4 tests, but now that you mention it I don't there is a reason to do so

Reply:
I'm happy if you want to deal with this in #157 if you're wary